crates/core_arch/src/x86/macros.rs

```diff
@@ -1,5 +1,51 @@
 //! Utility macros.
+// Helper struct used to trigger const eval errors when a const generic immediate value is
+// out of 2-bit range.
+pub(crate) struct ValidateConstImm2<const imm2: i32>;
+impl<const imm2: i32> ValidateConstImm2<imm2> {
+    pub(crate) const VALID: () = {
+        let _ = 1 / ((imm2 >= 0 && imm2 <= 3) as usize);
+    };
+}
+
+#[allow(unused)]
+macro_rules! static_assert_imm2 {
```

Comment:
Move these to the generic `macros.rs` instead of the x86-specific one.

Reply:
I wasn't exactly sure why some `constify_imm*` are duplicated here and there, so I added these new ones at the same places as the old ones. I'll combine this with the suggestion above.